tests/python/relay/strategy/arm_cpu/test_depthwise_conv2d.py

```diff
@@ -14,203 +14,125 @@
-                "-keys": "arm_cpu",
-                "-mcpu": "cortex-m7",
-            },
-            schedule_name=schedule_name,
-        )
-
-
-class TestDepthwiseConv2d_NCHW_OIHW(BasicDepthwiseConv2dTests):
+"""Tests for arm_cpu schedules for depthwise_conv2d."""
+
+from test_generalized_conv2d import GeneralizedConv2dTests
+from tvm.testing import parameter, parameters, main
+
+
+class DepthwiseConv2dTests(GeneralizedConv2dTests):
+    """Helper for constructing depthwise Conv2ds. Sets kernel layout to what x86 code supports."""
+
+    def setup_method(self):
+        self.ref_kernel_layout = "HWOI"
+
+
+class TestDepthwiseConv2d_NCHW_OIHW(DepthwiseConv2dTests):
     """This test is for depthwise_conv2d_nchw.arm_cpu schedule."""
 
-    data_shape, kernel_size, num_filter, strides, padding, dilation = tvm.testing.parameters(
-        ((1, 16, 32, 32), (3, 3), 16, 1, 0, 1),
-        ((1, 32, 10, 3), (3, 3), 32, 1, 0, 1),
-        ((1, 32, 32, 16), (3, 3), 32, 1, (0, 2, 2, 0), 1),
-        ((1, 32, 32, 16), (3, 3), 32, 1, 0, 1),
-        ((1, 32, 32, 16), (3, 3), 32, 1, 0, 1),
-        ((1, 32, 32, 16), (3, 3), 32, 1, (0, 2, 2, 0), 2),
-        ((1, 16, 32, 16), (3, 3), 16, 1, (1, 1, 2, 2), 2),
+    data_shape, groups, kernel_size, num_filter, strides, padding, dilation = parameters(
+        ((1, 32, 32, 16), 16, (3, 3), 16, 1, 0, 1),
+        ((1, 10, 3, 32), 32, (3, 3), 32, 1, 0, 1),
+        ((1, 32, 16, 32), 32, (3, 3), 32, 1, (0, 2, 2, 0), 1),
+        ((1, 32, 16, 32), 32, (3, 3), 32, 1, 0, 1),
+        ((1, 32, 16, 32), 32, (3, 3), 32, 1, 0, 1),
+        ((1, 32, 16, 32), 32, (3, 3), 32, 1, (0, 2, 2, 0), 2),
+        ((1, 32, 16, 16), 16, (3, 3), 16, 1, (1, 1, 2, 2), 2),
     )
-    data_layout = tvm.testing.parameter("NCHW")
-    dtype = tvm.testing.parameter("int8", "int16")
-    kernel_layout = tvm.testing.parameter("OIHW")
-    schedule_name = tvm.testing.parameter("depthwise_conv2d_nchw.arm_cpu")
+
+    in_dtype = parameter("int8", "int16")
+    data_layout = parameter("NCHW")
+    kernel_layout = parameter("OIHW")
+    out_layout = parameter("NCHW")
+    schedule_name = parameter("depthwise_conv2d_nchw.arm_cpu")
 
 
-class TestDepthwiseConv2d_NHWC_HWOI(BasicDepthwiseConv2dTests):
+class TestDepthwiseConv2d_NHWC_HWOI(DepthwiseConv2dTests):
     """This test is for depthwise_conv2d_nhwc.generic schedule."""
 
-    data_shape, kernel_size, num_filter, strides, padding, dilation = tvm.testing.parameters(
-        ((1, 32, 32, 16), (3, 3), 16, 1, 0, 1),
-        ((1, 32, 10, 16), (3, 3), 16, 1, 0, 1),
-        ((1, 49, 10, 64), (10, 4), 64, (2, 1), (4, 1, 5, 1), 1),
-        ((1, 32, 32, 16), (3, 3), 16, 1, (0, 2, 2, 0), 1),
-        ((1, 32, 32, 16), (3, 3), 16, 1, 0, 1),
-        ((1, 32, 32, 16), (3, 3), 16, 1, 0, 1),
-        ((1, 32, 32, 16), (3, 3), 16, 1, (0, 2, 2, 0), 2),
-        ((1, 32, 32, 16), (3, 3), 16, 1, (1, 1, 2, 2), 2),
+    data_shape, groups, kernel_size, num_filter, strides, padding, dilation = parameters(
```

Comment:
Being a `depthwise_conv2d` means that the number of groups equals the number of _input channels_, which might not equal the number of filters. Your point still stands tho - I've added a TVM fixture to avoid specifying the number of groups in each line.